File: src/main/kotlin/leetCode/easy/MissingNumber.kt
package leetCode.easy

class MissingNumber {
    fun missingNumber(nums: IntArray): Int {
        val set = nums.toSet()
        val n = nums.size + 1

        var i = 0
        while (i != n) {
            if (!set.contains(i)) return i
            i++
        }

        return 0
    }
}

fun main(args: Array<String>) {
    println(MissingNumber().missingNumber(intArrayOf(3, 0, 1)))
    println(MissingNumber().missingNumber(intArrayOf(0, 1)))
    println(MissingNumber().missingNumber(intArrayOf(9, 6, 4, 2, 3, 5, 7, 0, 1)))
}

File: go/leetCode/easy/MissingNumber.go
package easy

type void struct{}

var empty = void{}

func missingNumber(nums []int) int {
	set := make(map[int]void)

	for _, num := range nums {
		set[num] = empty
	}

	n := len(nums)

	for i := 0; i <= n; i++ {
		_, exists := set[i]
		if !exists {
			return i
		}
	}

	return -1
}
